File: IntersectionOfTwoArrays/InsecOfTwoArr.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        num1 = set(nums1)
        num2 = set(nums2)
        num1.intersection_update(num2)
        return num1


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        arr = []
        for i in nums1:
            if i in nums2:
                arr.append(i)
        return set(arr)


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        return {i for i in nums1 if i in nums2}


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        arr = []
        for i in nums1:
            left, right = 0, len(nums2) - 1
            while left <= right:
                mid = (left + right) // 2
                if nums2[mid] == i:
                    if i not in arr:
                        arr += i,
                    break
                elif nums2[mid] > i:
                    right = mid - 1
                else:
                    left = mid + 1
        return arr

File: IntersectionOfTwoArrays/InsecOfTwoArr.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  nums1 = new Set(nums1);
  return nums2.filter((num) => nums1.delete(num));
};


/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  let arr = [];
  nums1.forEach((num) => {
    if (nums2.indexOf(num) != -1) {
      arr.push(num);
    }
  });
  let res = [...new Set(arr)];
  return res;
};


/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  let hash = {};
  let arr = [];
  for (let num of nums1) {
    hash[num] = true;
  }
  for (let i = 0; i < nums2.length; i++) {
    if (hash[nums2[i]]) {
      arr.push(nums2[i]);
    }
  }
  return [...new Set(arr)];
};


/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  nums1 = nums1.sort((a, b) => a - b);
  nums2 = nums2.sort((a, b) => a - b);
  let arr = [];
  for (let i of nums1) {
    let left = 0;
    let right = nums2.length - 1;
    while (left <= right) {
      let mid = Math.ceil((left + right) / 2);
      if (nums2[mid] === i) {
        if (!arr.includes(i)) {
          arr.push(i);
        }
        break;
      } else if (nums2[mid] > i) {
        right = mid - 1;
      } else {
        left = mid + 1;
      }
    }
  }
  return arr;
};
